metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -141,7 +143,7 @@ class MetalsLanguageServer(
 
   private val fingerprints = new MutableMd5Fingerprints
   private val mtags = new Mtags
-  var workspace: AbsolutePath = _
+  var workspace: Workspace = _
```

Comment:
Do we need to change AbsolutePath to Workspace? That causes a bunch of unrelated changes

Reply:
I was thinking about adding and passing the boolean flag `isMetaWorkspace` through e.g. `BspConnector@connect`, `BuildServerConnection@fromSockets` etc. but that information is strictly coupled with the workspace path and separating them may lead to invalid states such as path pointing to the meta workspace and `isMetaWorkspace` being set to false. That's why I decided to do what I did. I agree that it causes a lot of changes, but for me, it seemed as a better option. What do you think?
Why that knowledge is needed at all? It's needed because metals server establishes one connection for project workspace and may establish more meta-connections for sbt meta-workspaces. Those meta connections are created later than workspace one and bsp trace file contains logs only from the last created connection. IMO only logs from the project workspace are valuable and some information about workspace is needed to decide whether setup tracing or not. https://github.com/scalameta/metals/blob/5de8af9708a173215babd55c89f04e4d9b4a6144/metals/src/main/scala/scala/meta/internal/metals/BuildServerConnection.scala#L268